tests/s25Main/integration/testAttacking.cpp

```diff
@@ -451,13 +462,75 @@ BOOST_FIXTURE_TEST_CASE(ConquerBld, AttackFixture<>)
     }
 }
 
+BOOST_FIXTURE_TEST_CASE(ArmoredSoldierLosesArmorInFight, AttackFixture<>)
+{
+    initGameRNG();
+
+    AddSoldiersWithRankAndArmor(milBld0Pos, 3, Job::General, 2);
+    AddSoldiersWithRankAndArmor(milBld1Pos, 1, Job::Private, 1);
+
+    // Finish recruiting, carrier outhousing etc.
+    RTTR_SKIP_GFS(400);
+
+    BOOST_TEST_REQUIRE(milBld0->GetNumTroops() == 3u);
+    BOOST_TEST_REQUIRE(milBld1->GetNumTroops() == 1u);
+
+    // Start attack ->1 (strong one first)
+    this->Attack(milBld1Pos, 1, true);
+    RTTR_EXEC_TILL(30, milBld0->GetNumTroops() == 2);
+
+    auto armoredSoldiers = [](nobMilitary* bld) {
```

Comment:
```suggestion
    auto countArmoredSoldiers = [](nobMilitary* bld) {
```

Similar below.  
My point is that there are multiple categories of things and the names need to reflect that, e.g.:
- numSoldiers (amount of)
- countSoldiers (get amount of)
- soldiers (actual soldiers)
- getSoldiers (retrieve)
- findSoldiers (expensive retrieve)